openedx/features/edly/tests/test_models.py

```diff
@@ -0,0 +1,91 @@
+"""
+Tests for Edly django models.
+"""
+import pytest
+from django.test import TestCase
+
+from openedx.features.edly.models import EdlyOrganization, EdlySubOrganization
+from openedx.features.edly.tests.factories import EdlyOrganizationFactory, EdlySubOrganizationFactory, SiteFactory
+
+
+class EdlyOrganizationTests(TestCase):
+    """
+    Tests for Organization Model.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(EdlyOrganizationTests, cls).setUpClass()
+
+    def test_organization_post_save_receiver(self):
+        """
+        Test organization model object creation.
+        """
+        org_name = 'test org name'
+        org_slug = 'testorgslug'
+        organization = EdlyOrganizationFactory(name=org_name, slug=org_slug)
+        organizations = EdlyOrganization.objects.filter(name=organization.name)
+        assert len(organizations) == 1
+
+    def test_organization_post_update_receiver(self):
+        """
+        Test Organization model object update.
+        """
+        org_name = 'updated test org name'
+        organization = EdlyOrganizationFactory()
+        organization.name = org_name
+        organization.save()
+
+        organizations = EdlyOrganization.objects.filter(name=org_name)
+
+        assert len(organizations) == 1
+        assert organizations[0].name == org_name
+
+    def test_slug_alphanumerice_validation(self):
+        org_slug = 'test $%3org slug'
+        organization = EdlyOrganizationFactory(slug=org_slug)
+        expect_exception = 'Only alphanumeric characters are allowed.'
+
+        with pytest.raises(Exception) as validation_exception:
+            organization.full_clean()
+
+        assert expect_exception in str(validation_exception.value)
+
+
+class EdlySubOrganizationTests(TestCase):
+    """
+    Tests for SubOrganization Model.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(EdlySubOrganizationTests, cls).setUpClass()
+        cls.lms_site = SiteFactory()
+        cls.studio_site = SiteFactory()
+
+    def test_sub_organization_post_save_receiver(self):
+        """
+        Test SubOrganization model object creation.
+        """
+        sub_organization = EdlySubOrganizationFactory.create(lms_site=self.lms_site)
+        sub_organization_data = EdlySubOrganization.objects.filter(lms_site=sub_organization.lms_site)
+        assert len(sub_organization_data) == 1
+
+    def test_sub_organization_post_update_receiver(self):
+        """
+        Test SubOrganization model object update.
```

Comment:
Still need to update names `SubOrganization`

Reply:
Done